### DISCUSSION TITLE
dolphinscheduler调度python提示权限不够

### DISCUSSION THREAD

**Discussion by echobigdata** [apache/dolphinscheduler#15085] [Q&A]
![image](https://github.com/apache/dolphinscheduler/assets/97231854/f6a71e74-7762-444d-be96-166827a2ec7b)

**Comment by EgbertHistorianPokling** [comment]
同问题，换root租客也不行，已放弃改成shell启动了。有别的解决方式的话我先插个眼

**Reply by HwiLu** [reply]
是个bug

**Reply by syedaqeel** [reply]
have we fixed it ?

**Reply by liuhen11** [reply]
可是就算shell启动，执行python的脚本的时候不会报权限不够的错误吗

**Reply by CHINAJAVABoy** [reply]
你执行的那个租户或者说是linux用户要有该目录的读写权限

**Comment by syedaqeel** [comment]
We have dolphin 3.2.0 and python 3.10 when we run a python task we get 
![dolphin permission denied on python task ](https://github.com/apache/dolphinscheduler/assets/16229721/23bead26-8179-430b-acab-535435da378f)

**Reply by HwiLu** [reply]
The solution could be found below
[https://asf-dolphinscheduler.slack.com/archives/C01L3LB9YMP/p1700795659818829](https://asf-dolphinscheduler.slack.com/archives/C01L3LB9YMP/p1700795659818829)

**Reply by ArkURL** [reply]
sry, but could you provide other link rather than slack? can't join dolphinscheduler's slack to read the solution.

**Reply by liuhen11** [reply]
me too!

**Reply by lxorc** [reply]
PYTHON_HOME is not set. set it in /etc/profile , look like 
```shell
# for 3.2.x
export PYTHON_LAUNCHER=/usr/local/bin/python3.9
# for 3.1.x and below
export PYTHON_HOME=/usr/local/bin/python3.9
```

**Reply by ArkURL** [reply]
thx, joined the team's slack to read to solutions just now👍
another solution is add env var to `/etc/profile`:
`echo "export PYTHON_LAUNCHER={your_python_path}" >> /etc/profile`

**Reply by lxorc** [reply]
set it in `/etc/profile`.

**Reply by Axiangg** [reply]
> ```shell
> export PYTHON_LAUNCHER=/usr/local/bin/python3.9
> ```

3.2.2 , make it , thanks

**Comment by lxorc** [comment]
PYTHON_HOME or PYTHON_LAUNCHER is not set. set it in /etc/profile , look like
```shell
# for 3.2.x
export PYTHON_LAUNCHER=/usr/local/bin/python3.9
# for 3.1.x and below
export PYTHON_HOME=/usr/local/bin/python3.9
```

**Comment by ygq010718** [comment]
确实就是python home的问题，我前天遇到之后试了试，加上这个就欧克了

**Reply by zui01** [reply]
如果运行用户和租户是一个用户会导致ds读取不到给他配置的python环境地址，只要给当前用户单独配置pythonpath即可

**Comment by CHINAJAVABoy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by dayjaby** [comment]
Still failing inside worker from docker-compose. Python is not even installed in the docker image.

**Reply by lxorc** [reply]
if you need run python script, you need install python first.

**Comment by CHINAJAVABoy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by HwiLu** [comment]
在ds启动用户.bashrc内加上Python路径
```
export PYTHON_LAUNCHER=/opt/ds_venv/bin/python
export PYTHON_HOME=/opt/ds_venv/bin/python
```

**Comment by CHINAJAVABoy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by lgb861213** [comment]
我利用docker compose部署之后也遇到了一样的错误日志，通过设置python环境变量参数之后依然运行失败，因为docker compose默认部署的worker没有安装PYTHON，所以还需要进行手动部署安装python

**Comment by Fortytwoo** [comment]
请确认两件事情：
1、PYTHON_HOME路径是否正确，如果不正确也会报无权限
2、脚本生成目录是否有权限，一般默认是/tmp路径下，初学者建议将权限开放所有人的读写执行权限

如果仍然不能执行，请开启开发者模式，这个模式下将保留生成的文件，这有助于你调试你的部署。

**Comment by abelsare348** [comment]
If you have installed it within docker container, then we will have to install python inside the docker container.

Follow the steps and work will be done:

1. log in inside the shell of container:
sudo docker exec -it dolphinscheduler /bin/bash

2. sudo apt install python

3. find out the python installation library
mostly usr/bin/python*version

4.create a environment in dolphinscheduler named PythonEnv
and give config
export PYTHON_LAUNCHER='usr/bin/python*version'

and give this env to python program and python program will start running on docker standalone dolphin scheduler environment

**Comment by CHINAJAVABoy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
